File: C++/LC1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i;
        vector<int> ans;
        map<int, int> position;
        
        if (nums.size() == 0) return ans;
        for (i = 0; i < nums.size(); i++) {
            auto iter = position.find(target - nums[i]);
			
            if (iter != position.end()) {
                ans.push_back(iter->second);
                ans.push_back(i);
                return ans;
            }
            position[nums[i]] = i;
        }
        return ans;
    }
};

File: Java/LC1.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> position = new HashMap<>();
        int i;
        
        for (i = 0; i < nums.length; i++) {
            int val = nums[i];
            
            if (position.containsKey(target - val)) {
                return new int[]{position.get(target - val).intValue(), i};
            }
            
            position.put(val, i);
        }
        
        return new int[0];
    }
}